starter-code/src/components/Modal.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import InputField from "./InputField";
+
+export default class Modal extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isActive: false,
+      modalClasses: "modal",
+      newFood:{},
+      message: false
+    };
+  }
+
+  openModal() {
+    this.setState({ isActive: !this.state.isActive });
+    this.setState({ modalClasses: this.state.modalClasses + " is-active" });
+  }
+
+  closeModal() {
+    this.setState({ isActive: !this.state.isActive });
+    this.setState({ modalClasses: "modal" });
+    this.setState({ message: false });
+  }
+
+  handleChange(input,name) {
+    this.state.message ? this.setState({ message: !this.state.message }) : null;
+    
+    this.state.newFood[name] = input;
+    this.setState({ newFood: this.state.newFood });
+    console.log(this.state.newFood)
+  }
+
+  handleAdd() {
+    if (
+      this.state.newFood.name === "" ||
```

Comment:
Te puedes ahorrar algo de codigo si haces una asignacion inicial del tipo const {name,calories,image} = this.state.newFood.